plugins/modules/azure_rm_adapplication.py

```diff
@@ -508,7 +500,11 @@ def __init__(self):
             key_value=dict(type='str', no_log=True),
             native_app=dict(type='bool'),
             oauth2_allow_implicit_flow=dict(type='bool'),
-            optional_claims=optional_claims_spec,
+            optional_claims=dict(
+                access_token=dict(type='list', elements='dict', options=claims_spec),
+                id_token=dict(type='list', elements='dict', options=claims_spec),
+                saml2_token=dict(type='list', elements='dict', options=claims_spec),
+                type=dict),
```

Comment:
optional_claims is currently a list of dictionaries, but as I mentioned in my original issue, this doesn't make sense because the claims can be applied to up to 3 different kinds of tokens. In addition to not working with the current codebase, optional_claims is ambiguous. Instead of a list, made it a dictionary with three sub-keys, one for each type of token. Each of these subkeys takes the same argument format that `optional_claims` current accepts.

There's an example module from another plugin [here](https://github.com/ansible-collections/cisco.nxos/blob/fdbfe11173aca042a47d96ddf62a069c24235f6b/plugins/modules/nxos_bgp_address_family.py#L39) that has a similar input format for part of its config